lib/src/domain/usecase/implementation/genres_usecase_impl.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:http/http.dart' as http;
+
+import '../../../core/util/states.dart';
+import '../../../data/datasource/remote/genres_api_service.dart';
+import '../../../data/repository/genre_repository_class.dart';
+import '../i_usecase.dart';
+
+class GenresUseCase implements IUseCase {
+  final GenresApiService apiService;
+  late GenreRepository repository;
```

Comment:
You can send directly the repository as a parameter of the useCase, and inside the constructor of the repository you can create the apiService

Reply:
solved